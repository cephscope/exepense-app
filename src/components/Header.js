import React from 'react';
import { Link, NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Fix The Bank</h1>
    <h3>Keep track of your expenses</h3>
    <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard!</NavLink> |
    <NavLink to="/create" activeClassName="is-active">Add Expense Page!</NavLink> |
  </header>
);

export default Header;