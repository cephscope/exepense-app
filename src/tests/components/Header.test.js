import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header'

 // only renders the given component, which is all that's necessary for this test suite

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
  
  // expect(wrapper.find('h1').text()).toBe('Expensify'); example enzyme shallow renderer usage
  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
});