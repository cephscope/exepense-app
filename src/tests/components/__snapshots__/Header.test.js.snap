// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Fix The Bank
  </h1>
  <h3>
    Keep track of your expenses
  </h3>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/dashboard"
  >
    Dashboard
  </NavLink>
   |
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Log New Expense
  </NavLink>
   |
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
